File: sampleandroid/src/main/java/com/example/sampleandroid/view/model/ApiSingleScreen.kt
package com.example.sampleandroid.view.model

import androidx.compose.Composable
import androidx.ui.foundation.Text
import androidx.ui.layout.Column
import androidx.ui.material.Button
import kim.jeonghyeon.androidlibrary.compose.widget.TextField
import kim.jeonghyeon.androidlibrary.extension.resourceToString
import kim.jeonghyeon.sample.compose.R
import kim.jeonghyeon.sample.viewmodel.ApiSingleViewModel

class ApiSingleScreen(private val model: ApiSingleViewModel = ApiSingleViewModel()) : ModelScreen(model) {
    override val title: String = R.string.single_call.resourceToString()

    @Composable
    override fun compose() {
        super.compose()
    }

    @Composable
    override fun view() {
        Column {
            Text("current value : ${+model.result}")
            TextField(model.input)
            Button(model::onClick) {
                Text("update")
            }
        }
    }
}

File: sample/sample-native/kotlinIOS/view/model/ApiSingleScreen.swift
//
// Created by hyun kim on 04/07/20.
// Copyright (c) 2020 hyun kim. All rights reserved.
//

import Foundation
import SwiftUI
import sample_base


struct ApiSingleScreen: SampleScreen {
    
    var model: ApiSingleViewModel = ApiSingleViewModel()
    
    func content(navigator: Navigator) -> some View {
        VStack(alignment: .center) {
            Text("current value : \(model.result.value!)")
            TextField("Enter value", text: +model.input).frame(width: 100, alignment: .center)
            Button(action: { self.model.onClick() }, label: { Text("Update")})
        }
        .navigationTitle("Api Single Call".localized())
    }
}

class ApiSingleScreen_Previews: PreviewProvider {
    static var previews: some View {
        ApiSingleScreen()
    }

}
